newton/_src/solvers/solver_data.py

```
@@ -0,0 +1,164 @@
+        for field in fields:
+            if hasattr(self, field):
+                continue
+            frequency = self.find_attribute_frequency(field)
+            field_size = self.frequency_sizes[frequency]
+
+            if self.verbose:
+                print(f"Initializing SolverData field {field} with size {field_size}")
+
+            hint = typing.get_type_hints(self).get(field)
+            if hint is None:
+                raise TypeError(f"SolverData field {field} is not defined.")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)